### PR TITLE
build: update `async-nats` to 0.33

### PR DESCRIPTION
**Pull request by rvolosatovs**
## Feature or Problem

Update `async-nats` to latest release

## Related Issues

This unblocks #724  (or at least it should)

### CODE REVIEW

**Review by vados-cosmonic — APPROVED** [review]
LGTM :rocket:

File: crates/provider-sdk/src/provider.rs [hunk 1]
```diff
@@ -274,15 +274,15 @@ impl ProviderConnection {
                                     if let Some(reply) = msg.reply {
                                         // send reply
                                         if let Err(err) = this.rpc_client
-                                            .publish_invocation_response(reply, resp).in_current_span().await {
+                                            .publish_invocation_response(reply.to_string(), resp).in_current_span().await {
```
Comment (connorsmith256) [inline crates/provider-sdk/src/provider.rs:277]: Why were these `to_string()` needed? Isn't `reply` a `Subject` already?

Comment (rvolosatovs) [reply crates/provider-sdk/src/provider.rs:277]: `reply` is `Subject` indeed, however `Subject` is a type defined in `async_nats` and therefore it does not belong in public API of this crate

Comment (connorsmith256) [reply crates/provider-sdk/src/provider.rs:277]: I don't think `publish_invocation_response` and `publish` should be public. They're part of the SDK's internal handling of invocations. Can we use the NATS type internally and avoid republishing? Normally I wouldn't care, but the RPC internals are where performance and allocations matter most, so it feels like a step backward to add these

Comment (rvolosatovs) [reply crates/provider-sdk/src/provider.rs:277]: We're dealing with network I/O here, so this really should not matter, but sure

Comment (rvolosatovs) [reply crates/provider-sdk/src/provider.rs:277]: `publish_invocation_response` is, indeed, public, so I am against this change

Comment (connorsmith256) [reply crates/provider-sdk/src/provider.rs:277]: I know, I'm saying these _shouldn't_ be public

Comment (rvolosatovs) [reply crates/provider-sdk/src/provider.rs:277]: Sure, but that change is out-of-scope for `async-nats` update PR, right?

**Review by thomastaylor312 — APPROVED** [review]

**Pull request merged**